File: Nian_Karl/SoftwareConstruction/src/org/uva/ql/ast/type/BoolType.java
package org.uva.ql.ast.type;

import org.uva.ql.visitor.TypeVisitor;

public class BoolType extends Type {

	@Override
	public boolean isBool() {
		return true;
	}
	
	@Override
	public boolean isEqual(Type type) {
		return type.isBool();
	}
	
	@Override
	public String toString() {
		return "Bool";
	}

	@Override
	public <T> T accept(TypeVisitor<T> visitor) {
		return visitor.visit(this);
	}
	
}

File: FelipezConde/QuestionnaireLanguage/AST/Types/BoolType.cs

namespace AST.Types
{
    public class BoolType : Type
    {
        public override bool IsBool()
        {
            return true;
        }

        public override bool IsEqual(Type type)
        {
            return type.IsBool();
        }

        public override T Accept<T>(Visitors.ITypeVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        public override string ToString()
        {
            return "bool";
        }

        public override Types.Type RetrieveType()
        {
            return this;
        }
    }
}
